identity/src/main/java/com/android/identity/DataTransportTcp.java

```diff
@@ -94,10 +94,15 @@ List<DataRetrievalAddress> parseDeviceRetrievalMethod(int version, @NonNull Data
         Map options = ((Map) items[2]);
 
         String host = Util.cborMapExtractString(options, RETRIEVAL_OPTION_KEY_ADDRESS);
-        int port = Util.cborMapExtractNumber(options, RETRIEVAL_OPTION_KEY_PORT);
+        long port = Util.cborMapExtractNumber(options, RETRIEVAL_OPTION_KEY_PORT);
+
+        if (port > 65535) {
```

Comment:
Might 0xFFFF be more readable? Up to you.

Reply:
I like 65535 :)

They're both sort of magic numbers. USHORT_MAX would be ideal, but I don't know of any such thing in Java.

Reply:
I think the right constant to use would be something like TCP_PORT_NUMBER_MAX but I don't know if there is such a constant.

Reply:
Not a bad idea - we could start by defining such a constant privately within this file, and then using it here?

private static final int TCP_PORT_NUMBER_MAX = 65535.

By the way, shouldn't we also check for port < 0? I.e., introduce a utility function

private static boolean isValidTcpPortNumber(long value) {
  return value >= 0 && value <= 65535;
}

and use that from here?

Reply:
I like the helper idea. Done.